src/analyzer/protocol/modbus/modbus-analyzer.pac

```diff
@@ -81,13 +81,20 @@ refine connection ModbusTCP_Conn += {
 		%{
 		return confirmed && orig_pdu && resp_pdu;
 		%}
+
+	function IsValidHeader( header: ModbusTCP_TransportHeader ) : bool
```

Comment:
@zambo99 - thanks for the PR. Could you see if the following patch does the same?

```
diff --git a/src/analyzer/protocol/modbus/modbus-protocol.pac b/src/analyzer/protocol/modbus/modbus-protocol.pac
index ebd1805815..5b566e6729 100644
--- a/src/analyzer/protocol/modbus/modbus-protocol.pac
+++ b/src/analyzer/protocol/modbus/modbus-protocol.pac
@@ -90,8 +90,8 @@ type ModbusTCP_PDU(is_orig: bool) = record {
 
 type ModbusTCP_TransportHeader = record {
        tid: uint16; # Transaction identifier
-       pid: uint16; # Protocol identifier
-       len: uint16; # Length of everything after this field
+       pid: uint16 &enforce(pid == 0); # Protocol identifier
+       len: uint16 &enforce(len >= 2); # Length of everything after this field
        uid: uint8;  # Unit identifier (previously 'slave address')
        fc:  uint8;  # MODBUS function code (see function_codes enum)
 } &byteorder=bigendian, &let {
```

This would trigger analyzer violations instead of skipping event raising if it wasn't a valid header. It should also result in appropriate analyzer.log / dpd.log entries.

Reply:
Indeed, with the two &enforce attributes added to the parser the effect is the same.
Your solution is more elegant. However, I think we should at least keep the btest I made in my patch to ensure a basic level of regression testing.
How shall we proceed? Do I make a new pull request?

Reply:
> However, I think we should at least keep the btest I made in my patch to ensure a basic level of regression testing.

Yeah, definitely.

> How shall we proceed? Do I make a new pull request?

You can also remove the commits with the `IsValidHeader()` addition and add a new commit with the enforce hunk. Then force-push.

You could consider adding `analyzer.log` to the baselines which should show the violations being raised.